cql2pgjson/src/test/java/org/z3950/zing/cql/cql2pgjson/ForeignKeyGenerationTest.java

```diff
@@ -22,28 +22,35 @@ public void ForeignKeySearchNumeric() throws FieldException, QueryValidationExce
     CQL2PgJSON cql2pgJson = new CQL2PgJSON("tablea.json" );
     cql2pgJson.setDbSchemaPath("templates/db_scripts/foreignKey.json");
     String sql = cql2pgJson.toSql("tableb.blah == /number 123452").getWhere();
-    assertEquals("tablea.id IN  ( SELECT Cast ( tableb.jsonb->>'tableaId'as UUID) from tableb WHERE (tableb.jsonb->>'blah')::NUMERIC = ('123452')::NUMERIC)", sql);
+    assertEquals("tablea.id IN  ( SELECT Cast ( tableb.jsonb->>'tableaId' as UUID) from tableb WHERE (tableb.jsonb->>'blah')::NUMERIC = ('123452')::NUMERIC)", sql);
```

Comment:
This space change shows that this kind of unit tests is hard to maintain.

Instead all unit tests in ForeignKeyGenerationTest.java that do not check thrown exceptions should be converted into 
ForeignKeyGenerationIT.java tests that check the functionality and not how the SQL in constructed.

Reply:
I am uncertain about this.  I still feel like these unit tests are useful to guide development and specify what sql is used.  Otherwise we would be debugging changes directly from the sql exceptions and syntax.  This seems like not that great a situation.  I used and am using these extensively to test my code, so why would they not be useful in the future?

Reply:
I understand. You need these test because subQuery method is too difficult to understand because you cannot see the complete SQL string, you can only see the complete SQL string in these tests.
However, then we need to fix subQuery method to be more easy to understand.
In this case most duplicate code was removed by extracting them into variables but readability suffered. Readability is more important than avoiding code duplication.

Two days ago I proposed this:

    if (tableNamePattern.matcher(term).matches()) {
      if (indexInTable) {
        return myField + " IN ( SELECT id FROM " + foreignTarget[0] + ")";
      }
      return myField + " IN ( SELECT Cast ( " + index2sqlText(foreignTableJsonb,  foreignTarget[1]) + " as UUID)" 
            + " FROM " + foreignTarget[0] + ")";
    }

This can be combined with splitting into a parentChild and a childParent method that handle the two indexInTable cases.